provider/dynamodb.go

```diff
@@ -281,12 +281,16 @@ func (store *dynamodbOnlineStore) CheckHealth() (bool, error) {
 		return false, fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 	}
 	if len(listOutput.TableNames) == 0 {
-		return false, NewProviderError(Connection, pt.DynamoDBOnline, Ping, "no tables found")
+		wrapped := fferr.NewConnectionError(pt.DynamoDBOnline.String(), fmt.Errorf("no tables found"))
+		wrapped.AddDetail("action", "ping")
+		return false, wrapped
 	}
 	scanInput := &dynamodb.ScanInput{TableName: listOutput.TableNames[0], Limit: aws.Int64(1)}
 	_, err = store.client.Scan(scanInput)
 	if err != nil {
-		return false, NewProviderError(Connection, pt.DynamoDBOnline, Ping, err.Error())
+		wrapped := fferr.NewConnectionError(pt.DynamoDBOnline.String(), fmt.Errorf("no tables found"))
```

Comment:
Feel like this is also a misdirection. Shouldnt we just check to see if the query failed on the ListTables?